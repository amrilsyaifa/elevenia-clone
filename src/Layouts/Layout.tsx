import React from 'react';
import { Layout as AntdLayout } from 'antd';
import 'antd/dist/antd.min.css';
import PageHeader from 'src/Reusables/Components/PageHeader';
import Style from './Layout.module.scss';

const { Content } = AntdLayout;

interface ILayout {
  children: React.ReactNode;
}

const Layout: React.FC<ILayout> = ({ children }) => {
  return (
    <div className={Style['layout-content']}>
      <AntdLayout>
        <div className={Style['page-header-container']}>
          <div className={Style['page-header']}>
            <PageHeader />
          </div>
        </div>
        <div className={Style['container']}>
          <Content style={{ padding: '0 50px' }}>
            <div className={Style['site-layout-content']}>{children}</div>
          </Content>
        </div>
      </AntdLayout>
    </div>
  );
};

export default Layout;
