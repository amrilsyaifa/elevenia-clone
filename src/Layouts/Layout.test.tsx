import { render, screen } from '@testing-library/react';
import React from 'react';
import Layout from './Layout';

describe('Layout', () => {
  test('Layout Render Snapshoot', () => {
    const { container } = render(
      <Layout>
        <div>render layout</div>
      </Layout>
    );
    expect(container).toMatchSnapshot();
  });

  test('PageHeader Have Children render correctly', () => {
    const childrenText = 'render layout';
    render(
      <Layout>
        <div>{childrenText}</div>
      </Layout>
    );
    const titleHeader = screen.getByText(childrenText);
    expect(titleHeader).toBeInTheDocument();
  });
});
