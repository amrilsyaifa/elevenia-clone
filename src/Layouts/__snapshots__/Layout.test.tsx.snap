// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Layout Render Snapshoot 1`] = `
<div>
  <div
    class="layout-content"
  >
    <section
      class="ant-layout"
    >
      <div
        class="container"
      >
        <div
          class="ant-page-header site-page-header ant-page-header-ghost"
        >
          <div
            class="ant-page-header-heading"
          >
            <div
              class="ant-page-header-heading-left"
            >
              <span
                class="ant-page-header-heading-title"
                title="Title"
              >
                Title
              </span>
              <span
                class="ant-page-header-heading-sub-title"
                title="This is a subtitle"
              >
                This is a subtitle
              </span>
            </div>
          </div>
        </div>
        <main
          class="ant-layout-content"
          style="padding: 0px 50px;"
        >
          <div
            class="site-layout-content"
          >
            <div>
              render layout
            </div>
          </div>
        </main>
      </div>
    </section>
  </div>
</div>
`;
